rust/parser/mod.rs

```diff
@@ -127,13 +127,12 @@ fn parse_single(rule: Rule, string: &str) -> Result<Node<'_>> {
 }
 
 pub(crate) fn visit_query_prefix(string: &str) -> Result<(Query, usize)> {
-    let parsed = parse_single(Rule::eof_query_prefix_partial, string);
+    let parsed = parse_single(Rule::eof_query_prefix, string);
     match parsed {
         Ok(node) => {
             let mut children = node.into_children();
             let query = children.consume_expected(Rule::query);
-            let remaining = children.consume_expected(Rule::any_partial);
-            let end_of_query_index = remaining.span().unwrap().begin_offset;
+            let end_of_query_index = query.span().unwrap().end_offset;
```

Comment:
The problem is @dmitrii-ubskii  that we revert back to not being able to include comments and whitespace after. But I think we can live with that...